packages/contracts/test/DatabaseResolver.t.sol

```diff
@@ -1,61 +1,49 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.13;
 
-import "forge-std/console.sol";
 import {Test} from "forge-std/Test.sol";
-import "@ens-contracts/registry/ENSRegistry.sol";
+import {ENSRegistry} from "@ens-contracts/registry/ENSRegistry.sol";
 import {
     PublicResolver,
     INameWrapper
 } from "@ens-contracts/resolvers/PublicResolver.sol";
 import "@ens-contracts/reverseRegistrar/ReverseRegistrar.sol";
 import "@ens-contracts/utils/UniversalResolver.sol";
 
-import "../script/Helper.sol";
-import {Config} from "../script/Config.s.sol";
+import {ENSHelper} from "../script/Helper.sol";
+import {DatabaseConfig} from "../script/config/DatabaseConfig.s.sol";
 import {DatabaseResolver} from "../src/DatabaseResolver.sol";
 import {DatabaseResolverScript} from "../script/deploy/DatabaseResolver.s.sol";
 
 contract DatabaseResolverTest is Test, ENSHelper {
 
     DatabaseResolver public resolver;
-    ENSRegistry public registry;
+    ENSRegistry registry;
     address owner;
 
     // Initial setup before each test
     function setUp() public {
         owner = address(this);
-        Config config = new Config(block.chainid);
+        DatabaseConfig config = new DatabaseConfig(block.chainid);
         (
             string memory gatewayUrl,
             uint256 gatewayTimestamp,
-            address[] memory signers
+            address[] memory signers,
+            ENSRegistry _registry
         ) = config.activeNetworkConfig();
+        registry = _registry;
         string[] memory urls = new string[](1);
         urls[0] = gatewayUrl;
 
-        registry = new ENSRegistry();
-
-        new UniversalResolver(address(registry), urls);
-        ReverseRegistrar registrar = new ReverseRegistrar(registry);
-
-        // .reverse
-        registry.setSubnodeOwner(rootNode, labelhash("reverse"), owner);
-        // addr.reverse
-        registry.setSubnodeOwner(
-            namehash("reverse"), labelhash("addr"), address(registrar)
-        );
-
         resolver = new DatabaseResolver(gatewayUrl, gatewayTimestamp, signers);
-
-        registrar.setDefaultResolver(address(resolver));
     }
 
     // Test the resolver setup from the constructor
-    function test_ResolverSetupFromConstructor() public {
-        Config config = new Config(block.chainid);
-        ( /* gatewayUrl */ , /* gatewayTimestamp */, address[] memory signers) =
-            config.activeNetworkConfig();
+    function testResolverSetupFromConstructor() public {
+        DatabaseConfig config = new DatabaseConfig(block.chainid);
+        ( /* gatewayUrl */
```

Comment:
why does it have this commented parameters? Is it a type of patter?

Reply:
yep, it kind of an [array destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) as JS does